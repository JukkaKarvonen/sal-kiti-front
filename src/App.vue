<template>
  <div>
    <nav>
      <TheNavbar />
    </nav>
    <main>
      <b-container>
        <router-view />
      </b-container>
    </main>
  </div>
</template>

<script>
import TheNavbar from "@/components/TheNavbar.vue";
export default {
  name: "app",
  components: {
    TheNavbar
  },
  mounted() {
    this.$store.dispatch("userInfo");
  },
  created() {
    document.title = this.$t("title.prefix");
  }
};
</script>
<style lang="scss">
@import "../node_modules/bootstrap/scss/bootstrap.scss";
@import "../node_modules/bootstrap-vue/src/index.scss";
@import "assets/sjal.scss";
</style>
