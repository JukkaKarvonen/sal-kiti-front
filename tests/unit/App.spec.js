import { shallowMount, createLocalVue } from "@vue/test-utils";
import App from "@/App.vue";
import VueRouter from "vue-router";

import BootstrapVue from "bootstrap-vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.use(VueRouter);
localVue.use(Vuex);

describe("App.vue", () => {
  let actions = {
    userInfo: jest.fn()
  };
  const store = new Vuex.Store({ actions });
  it("renders content", () => {
    const wrapper = shallowMount(App, {
      localVue,
      store
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(actions.userInfo).toHaveBeenCalled();
  });
});
