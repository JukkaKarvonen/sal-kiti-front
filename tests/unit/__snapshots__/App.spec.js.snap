// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders content 1`] = `
<div>
  <nav>
    <thenavbar-stub />
  </nav>
   
  <main>
    <b-container-stub
      tag="div"
    >
      <router-view-stub
        name="default"
      />
    </b-container-stub>
  </main>
</div>
`;
